libnvme-rs/src/nvme_tree.rs

```diff
@@ -0,0 +1,149 @@
+/// RAII Wrapper for nvme_root_t
+pub struct NvmeRoot {
+    root: *mut crate::bindings::nvme_root,
+}
+
+impl NvmeRoot {
+    pub fn new(root: *mut crate::bindings::nvme_root) -> Self {
+        NvmeRoot {
+            root,
+        }
+    }
+    pub fn as_mut_ptr(&self) -> *mut crate::bindings::nvme_root {
```

Comment:
The goal of the lib should be to NOT have to deal with these raw dangling pointers, why is this method here and public?

Reply:
That's an oversight, it's meant to be `pub(crate)`, will fix.